### ISSUE TITLE
"rs" while waiting for file change doesn't work

### ISSUE THREAD

**Issue by iliakan** [remy/nodemon#398]
When the process dies, the output is:
```
[nodemon] app crashed - waiting for file changes before starting...
```

Sometimes it would be nice if nodemon restarted the app manually (after a second) or at least I could type `rs` for it. 

Just because file changes are not needed / already done (happens too).

As of now, none of these 2 possibilities work. 
A feature request would be to implement any of them.

**Comment by remy** [comment]
Can people give this another test is nodemon 1.3.x? Would be good to know if this affects people.

**Comment by remy** [comment]
I'm resurrecting this issue because I've seen an issue with node 0.12.0 and the `rs` restart.

If you've had this issue, can you confirm your OS, node version and nodemon version?

**Comment by iliakan** [comment]
Confirming, can reproduce.
MacOS Yosemite latest.

```
iliakan> node -v
v0.12.0

[gulp] [nodemon] v1.3.6
[gulp] [nodemon] to restart at any time, enter `rs`
[gulp] [nodemon] watching: /js/javascript-nodejs/handlers/**/* /js/javascript-nodejs/modules/**/*
[gulp] [nodemon] starting `node --debug --harmony ./bin/server`
Debugger listening on port 5858
currentLng set to: ru
[20:09:58] livereload: deferred listen on change public/{i,img,js,styles}/**/*.*
loaded file: locales/ru/translation.json
17:09:59.410Z  INFO app: App listening 0.0.0.0:3000
app is listening
[20:10:01] [webpack] Time: 3989ms
                           Asset     Size  Chunks             Chunk Names
  search.js?c43f4713106b59894299     3974       0  [emitted]  search
  getpdf.js?c43f4713106b59894299    27307       1  [emitted]  getpdf
  footer.js?c43f4713106b59894299    28565       2  [emitted]  footer
 profile.js?c43f4713106b59894299   312107       3  [emitted]  profile
 angular.js?c43f4713106b59894299  3043907       4  [emitted]  angular
tutorial.js?c43f4713106b59894299   182228       5  [emitted]  tutorial
       6.js?c43f4713106b59894299     5266       6  [emitted]  ebook
    head.js?c43f4713106b59894299   137568       7  [emitted]  head
       8.js?c43f4713106b59894299   140062       8  [emitted]  authClient
[20:10:02] /js/javascript-nodejs/public/js/tutorial.js reloaded.
[gulp] [nodemon] app crashed - waiting for file changes before starting...
rs
rs
rs
rs
^C
```

**Comment by tivac** [comment]
I can also confirm that after an app crash `rs` doesn't restart the process, I have to touch a file.

Win 7 x64
iojs 1.5.1 x64
nodemon 1.3.7

**Comment by danmartinez101** [comment]
I can confirm this also. Additionally, it appears that it is queuing the `rs` commands as evidenced below.

```

> haakon-core@0.0.1 tdd /Users/dmartinez/source/workspace/haakon-core
> nodemon ./node_modules/.bin/jasmine

23 Mar 17:57:18 - [nodemon] v1.3.7
23 Mar 17:57:18 - [nodemon] to restart at any time, enter `rs`
23 Mar 17:57:18 - [nodemon] watching: *.*
23 Mar 17:57:18 - [nodemon] starting `node ./node_modules/.bin/jasmine`
Failed to load c++ bson extension, using pure JS version
Started
...................*...........

Pending:

1) ContactResource delete instantiates and executes a DeleteContactCommand
  not yet implemented

31 specs, 0 failures, 1 pending spec
Finished in 0.037 seconds
23 Mar 17:57:18 - [nodemon] clean exit - waiting for changes before restart
rs

rs
rs
rs
23 Mar 17:59:21 - [nodemon] restarting due to changes...
23 Mar 17:59:21 - [nodemon] starting `node ./node_modules/.bin/jasmine`
23 Mar 17:59:21 - [nodemon] starting `node ./node_modules/.bin/jasmine`
23 Mar 17:59:21 - [nodemon] starting `node ./node_modules/.bin/jasmine`
23 Mar 17:59:21 - [nodemon] starting `node ./node_modules/.bin/jasmine`
23 Mar 17:59:21 - [nodemon] starting `node ./node_modules/.bin/jasmine`
Failed to load c++ bson extension, using pure JS version
Started
...............................


31 specs, 0 failures
Finished in 0.043 seconds
23 Mar 17:59:22 - [nodemon] clean exit - waiting for changes before restart

```

**Comment by danmartinez101** [comment]
I had recently update node to the 12.1 release. I just tried this again after downgrading and that seems to have made the issue go away.

**Comment by remy** [comment]
Can you give this a go with latest nodemon (currently 1.7) - I found a bug with iojs (and node 4) which is fixed now.

**Comment by danmartinez101** [comment]
I'll let you know tonight

**Comment by iliakan** [comment]
That really looked like node.js bug. Will trace too.

**Comment by danmartinez101** [comment]
It works in both 0.12.x and 4.0 

Yay :+1:

**Comment by roblav96** [comment]
I'm still getting this issue here :/

**Comment by remy** [comment]
Please debug?

**Comment by roblav96** [comment]
Hey sorry, I thought I left an update on this :X 

I'm using https://github.com/SocketCluster/socketcluster
The worker and broker processes would crash on 'SIGUSR2' so added this hook to make it work:
```
process.on( 'SIGUSR2', function () {
	socketCluster.killWorkers()
	socketCluster.killBrokers()
	process.exit(0)
} )
```
It works great! Thank you!

**Comment by rotsee** [comment]
This bug still affects me on `nodemon@1.18.4` (Ubuntu 18.04).

No Gulp or Grunt or anything else involved, just running `nodemon` from the command line and Nodemon will not listen to `rs` after a crash. (As far as I can remember this has always been the behaviour, and I have tried rolling back the version as far as 1.0.13, with no effect)

**Comment by pearofducks** [comment]
Still happening here on multiple platforms vs. `nodemon@1.18.7`

Reproduction: https://hub.docker.com/r/pearofducks/nodemon-repro/

1. Start the container: `docker run -it pearofducks/nodemon-repro sh`
2. Start a very basic Koa app: `cd /app && yarn dev`
3. Start another shell: `docker exec -it HASH_OF_OTHER_CONTAINER sh`
4. Try to start the koa app again from 2, app crashes since port is taken, but `rs` doesn't work
